bulk_indexer.go

```diff
@@ -709,14 +729,22 @@ func indexnth(s []byte, nth int, sep rune) int {
 	})
 }
 
-type errorFlushFailed struct {
+type ErrorFlushFailed struct {
```

Comment:
this doesn't have to be exported now imo.

Reply:
I commented because it isn't necessary in this PR, and it increases the public interface that we need to maintain in this lib. I'm fine with your current change.